libsolidity/ast/TypeProvider.cpp

```diff
@@ -233,7 +233,11 @@ Type const* TypeProvider::fromElementaryTypeName(ElementaryTypeNameToken const&
 	case Token::UFixed:
 		return fixedPoint(128, 18, FixedPointType::Modifier::Unsigned);
 	case Token::Address:
+	{
+		if (_stateMutability && *_stateMutability == StateMutability::Payable)
```

Comment:
I think you can actually do `_stateMutability == StateMutability::Payable` but it might be clearer this way.

Reply:
Changed it to assert anyway